File: leetcode/122/MaxProfit.js
var maxProfit = function(prices) {
    if(prices.length == 0)
        return 0;
    
    let profit = 0;
    
    for(let i = 0; i < prices.length -1; i++)
    {
        if(prices[i+1] > prices[i])
        {
            profit += prices[i+1] - prices[i];
        }
    }
    
    return profit;
};

console.log(maxProfit([7, 1, 5, 3, 6, 4 ]));

File: leetcode/122/MaxProfit.cs
public class Solution
{
    public int MaxProfit(int[] prices)
    {
        if (prices.Length < 2)
            return 0;

        int profit = 0;

        for (int i = 0; i < prices.Length - 1; i++)
        {
            if (prices[i + 1] > prices[i])
            {
                profit += prices[i + 1] - prices[i];
            }
        }

        return profit;
    }
}